tools/make-single-file.js

```diff
@@ -44,12 +45,108 @@ const presets = {
     },
 }
 
-const preset = process.argv[2] || 'dist'
+const {values: {preset, out: outdir, story_file: story_file_path}} = util.parseArgs({options: {
+    "preset": {type: "string", default: "dist"},
+    "story_file": { type: "string" },
+    "out": { type: "string", default: path.join(webpath, '../single-file') },
+}})
+
 if (!presets[preset]) {
-    throw new Error(`Unknown preset: ${process.argv[2]}`)
+    throw new Error(`Unknown preset: ${preset}`)
 }
 const options = Object.assign({}, base_options, presets[preset])
 
+if (story_file_path) {
+    try {
+        options.story_file = await fs.readFile(story_file_path)
+    } catch (cause) {
+        throw new Error(`Couldn't read story_file_path ${story_file_path}`, {cause})
+    }
+    // this is a stripped-down version of src/common/formats.js, so we don't have to import the world
+    const formats = [
+        {
+            id: 'blorb',
+            extensions: /\.(blb|blorb)/i,
+        },
+        {
+            id: 'adrift4',
+            extensions: /\.taf/i,
+            engines: [
```

Comment:
This structure doesn't need to match the one from formats, so it could just have a `engine` property with the single id we want to include.